### ISSUE TITLE
mongo 4.4 binary

### ISSUE THREAD

**Issue by n6g7** [themattman/mongodb-raspberrypi-binaries#6]
I am looking for raspberrypi compatible mongodb 4.4 binary / docker image as, unfortunately, that is the [highest version supported by UniFi network app](https://help.ui.com/hc/en-us/articles/115005159588-UniFi-Tuning-the-Network-Application-for-a-High-Number-of-UniFi-Devices#4) (cf. section "Changing Mongo Version/Engine").
Is that a version that you're interested in providing in this repo? If not, can you comment on the likelihood that the steps in the readme would work for that version? I might try to build it myself if necessary.

**Comment by themattman** [comment]
It's not in my plans as I hope to encourage the community to use newer releases.

However, because of the unique motivation you describe behind the 4.4 release I may consider a special release for it in the future, potentially near the EOL date. My immediate plans are to support the Pi 5 and other various items.

I have never built a 4.4 binary from source, and the build steps do vary between major version releases. I would venture to guess that these instructions would need a fair amount of tinkering to work two major versions back. If you are interested in doing it yourself, I would recommend starting with the `docs/` dir of the relevant branch of `mongodb/mongo` for build instructions.

**Comment by GeoffWilliams** [comment]
Hello! thanks for making this repo.

I needed mongo 4.4(.26) too so I was able to adapt the instructions to make a package. Would be great if this could be adopted or published here:

The build script: https://github.com/GeoffWilliams/mongodb-raspberrypi-binaries/blob/mongo4.4/mongo-4.4-build.sh
**The packages:  https://github.com/GeoffWilliams/mongodb-raspberrypi-binaries/releases/tag/v4.4.26**

Why the heck am I packaging 4.4??? Because its needed by TP Link Omada software (requires < 4.5)  and also ubiquity I believe for the same reason.

Would be great if these companies could update their mongos but it is what it is really.

Only the instructions for ubuntu focal work. I tried jammy but my compiler flags were broken or something...

**Comment by themattman** [comment]
Thanks for your contribution @GeoffWilliams! I'll try to incorporate your work over the next few weeks into this repo and the [docker images](https://github.com/themattman/mongodb-raspberrypi-docker) repo too. Thanks for the heads up on your use case as well.

**Comment by eoprede** [comment]
There are multiple reports that at least unifi network appliance works just fine with latest MongoDB versions. I can confirm it's running with 7.0.5 without any issues that I can spot.

**Comment by FeatherKing** [comment]
@eoprede fair point, im using this 4.4 binary, but maybe ill make the move to 7 if its working that well

**Comment by pmkane** [comment]
I'd also love to see 4.4 tracked, for the same reasons (unifi support on pi4).

**Comment by themattman** [comment]
Now that a new release - [4.4.28](https://www.mongodb.com/docs/manual/release-notes/4.4/#4.4.28---jan-18--2024) - was cut, I'll work on this in the upcoming week. AFAIK there will be one more release before [this version goes EOL](https://www.mongodb.com/legal/support-policy/lifecycles) so there shouldn't be a need to set up tracking on this branch.

**Comment by GeoffWilliams** [comment]
This is great, thanks so much :beers:  I'll use this in the next release of [omadapi](https://omadapi.com/) ... I packaged the whole system up like retropi

**Comment by RicardoM17** [comment]
Sorry to hijack this thread but is a `5.0.25` binary in the works? Unfortunately we got stuck with having to update Mongo in our systems but `5.0.25` is not supported because they are below ARMv8.2...

**Comment by themattman** [comment]
I wasn't planning on supporting 5.0. Can you provide more detail? Is a newer version like 6.x or 7.x not feasible to use?

**Comment by themattman** [comment]
Closing as Won't Do.
v7.0+ should be sufficient these days and has more features 😄  (E.g. Omadapi [was updated](https://github.com/GeoffWilliams/omadapi/issues/2#issuecomment-2799935573) to support v7)
